Vagrantfile

```diff
@@ -0,0 +1,121 @@
+
+  # Create a public network, which generally matched to bridged network.
+  # Bridged networks make the machine appear as another physical device on
+  # your network.
+  # config.vm.network "public_network"
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # By default vagrant sync the current directory. Let's disabled it because the directory
+  # will be synced later to the proper destination.
+  config.vm.synced_folder ".", "/vagrant", disabled:true
+  # TODO: Ideally we should be using nfs or 9p for bidirectional syncing
+  config.vm.synced_folder ".", "/home/vagrant/go/src/github.com/kata-containers/tests", type:"rsync"
+  config.vm.synced_folder "#{ENV['GOPATH']}/src/github.com/kata-containers/kata-containers",
+  	"/home/vagrant/go/src/github.com/kata-containers/kata-containers", type:"rsync"
+
+  # Provider-specific configuration so you can fine-tune various
+  # backing providers for Vagrant. These expose provider-specific options.
+  # Example for VirtualBox:
+  #
+  # config.vm.provider "virtualbox" do |vb|
+  #   # Display the VirtualBox GUI when booting the machine
+  #   vb.gui = true
+  #
+  #   # Customize the amount of memory on the VM:
+  #   vb.memory = "1024"
+  # end
+  config.vm.provider "libvirt" do |lv|
+    lv.driver = "kvm"
+    lv.cpus = "4"
+    lv.memory = "8192"
+  end
+
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  config.vm.provision "shell", reboot: true, env: {"CI_JOB" => job},
+                      inline: <<-SHELL
+
+    export GOPATH="/home/vagrant/go"
+    kata_tests_repo_dir="${GOPATH}/src/github.com/kata-containers/tests"
+
+    env_file="/home/vagrant/ci_job_env"
+    cat <<-EOF > ${env_file}
+export GOPATH="$GOPATH"
+export PATH="/usr/local/go/bin:\$GOPATH/bin:\$PATH"
+export CI="true"
+export CI_JOB="$CI_JOB"
+pushd $kata_tests_repo_dir
+source .ci/ci_job_flags.sh
+popd
+EOF
+
+    source ${env_file}
+
+    # Customize the .bashrc so that it will have the variables exported
+    # after log-in.
+    echo "cd $kata_tests_repo_dir" >> /home/vagrant/.bashrc
+    echo "source $env_file" >> /home/vagrant/.bashrc
+
+    # FIXME: The kubernetes tests. Maybe it should be handled on
+    # integration/kubernetes/init.sh
+    systemctl stop firewalld || true
+    swapoff -a || true
+
+    # Fedora >= 32 Kernel comes configured with cgroup v2 by default.
+    # This switches back to cgroup v1. It requires a reboot.
+    sudo dnf install -y grubby
```

Comment:
Although you've specified a Fedora "box" in the config, it prolly wouldn't hurt to source `/etc/os-release` and die unless you find a Fedora-like system.

Reply:
I am working on a multi-distro version of this vagrantfile so I will have to use the `/etc/os-release` anyway :)